[bullet3] Add support for BulletRobotics & Extras

****

**Is your feature request related to a problem? Please describe.**
The current port explicitly disables building and linking the Extras(-DBUILD_EXTRAS), which means the BulletRobotics code is not built, which cuts off a relatively big chunk of the library, and makes it significantly harder to use.

**Proposed solution**
Configure the library to optionally build BulletRobotics and other Extras.

**Describe alternatives you've considered**
Building outside vcpkg is a relatively painful solution, one that often results in hard to maintain projects.

**Additional context**
Look at the difference in complexity between the InverseKinematics example and the RobotSimulator/HelloBulletRobotics example.

The error like this:
```
1>bullet3Project.obj : error LNK2019: unresolved external symbol "public: __cdecl b3RobotSimulatorClientAPI_NoGUI::b3RobotSimulatorClientAPI_NoGUI(void)" (??0b3RobotSimulatorClientAPI_NoGUI@@QEAA@XZ) referenced in function main
1>bullet3Project.obj : error LNK2019: unresolved external symbol "public: virtual __cdecl b3RobotSimulatorClientAPI_NoGUI::~b3RobotSimulatorClientAPI_NoGUI(void)" (??1b3RobotSimulatorClientAPI_NoGUI@@UEAA@XZ) referenced in function main
1>C:\Users\v-ruil\source\repos\bullet3Project\x64\Debug\bullet3Project.exe : fatal error LNK1120: 2 unresolved externals
1>Done building project "bullet3Project.vcxproj" -- FAILED.
```
It seems that `b3RobotSimulatorClientAPI_NoGUI `is from the source code listed in `example  `folder.
I removed `b3RobotSimulatorClientAPI_NoGUI test3;`, the project can build successfully.

I also tried to build example and set 
```
-DBUILD_BULLET2_DEMOS=ON
-DBUILD_BULLET3=ON 
```
but there were still some errors like this:
```
LINK : warning LNK4075: ignoring '/INCREMENTAL' due to '/OPT:ICF' specification
   Creating library examples\ExampleBrowser\App_ExampleBrowser.lib and object examples\ExampleBrowser\App_ExampleBrowser.exp
Error copying directory from "F:/test/vcpkg/buildtrees/bullet3/src/2.89-eb26871ed5/data" to "F:/test/vcpkg/buildtrees/bullet3/x64-windows-static-dbg/data".
[902/903] cmd.exe /C "cd . && F:\test\vcpkg\downloads\tools\cmake-3.14.0-windows\cmake-3.14.0-win32-x86\bin\cmake.exe -E vs_link_exe --intdir=examples\SharedMemory\CMakeFiles\App_PhysicsServer_SharedMemory_VR.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100183~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100183~1.0\x64\mt.exe --manifests  -- "E:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.25.28610\bin\Hostx64\x64\link.exe"  @CMakeFiles\App_PhysicsServer_SharedMemory_VR.rsp  /out:examples\SharedMemory\App_PhysicsServer_SharedMemory_VR.exe /implib:examples\SharedMemory\App_PhysicsServer_SharedMemory_VR.lib /pdb:examples\SharedMemory\App_PhysicsServer_SharedMemory_VR.pdb /version:0.0  /machine:x64 /OPT:REF /OPT:ICF /INCREMENTAL:NO /debug /INCREMENTAL /subsystem:console  && cmd.exe /C "cd /D F:\test\vcpkg\buildtrees\bullet3\x64-windows-static-dbg\examples\SharedMemory && F:\test\vcpkg\downloads\tools\cmake-3.14.0-windows\cmake-3.14.0-win32-x86\bin\cmake.exe -E copy_if_different F:/test/vcpkg/buildtrees/bullet3/src/2.89-eb26871ed5/examples/ThirdPartyLibs/openvr/bin/win64/openvr_api.dll F:/test/vcpkg/buildtrees/bullet3/x64-windows-static-dbg/examples/SharedMemory/openvr64pi.dll && cd /D F:\test\vcpkg\buildtrees\bullet3\x64-windows-static-dbg\examples\SharedMemory && F:\test\vcpkg\downloads\tools\cmake-3.14.0-windows\cmake-3.14.0-win32-x86\bin\cmake.exe -E copy_directory F:/test/vcpkg/buildtrees/bullet3/src/2.89-eb26871ed5/data F:/test/vcpkg/buildtrees/bullet3/x64-windows-static-dbg/data""
Microsoft (R) Incremental Linker Version 14.25.28612.0
```
I'm not sure why the files cannot be copied successfully.

--- a/ports/bullet3/portfile.cmake
+++ b/ports/bullet3/portfile.cmake
@@ -15,6 +15,7 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
     FEATURES
         multithreading       BULLET2_MULTITHREADING
         double-precision     USE_DOUBLE_PRECISION
+		extras               BUILD_EXTRAS
     INVERTED_FEATURES
         rtti                 USE_MSVC_DISABLE_RTTI
 )
@@ -27,7 +28,6 @@ vcpkg_cmake_configure(
         -DBUILD_BULLET2_DEMOS=OFF
         -DBUILD_OPENGL3_DEMOS=OFF
         -DBUILD_BULLET3=OFF
-        -DBUILD_EXTRAS=ON
         -DBUILD_BULLET_ROBOTICS_GUI_EXTRA=OFF
         -DBUILD_BULLET_ROBOTICS_EXTRA=OFF
         -DBUILD_GIMPACTUTILS_EXTRA=OFF        

--- a/ports/bullet3/portfile.cmake
+++ b/ports/bullet3/portfile.cmake
@@ -1,5 +1,4 @@
 vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
-vcpkg_minimum_required(VERSION 2022-10-12) # for ${VERSION}
 
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH

--- a/ports/bullet3/portfile.cmake
+++ b/ports/bullet3/portfile.cmake
@@ -14,7 +14,7 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
     FEATURES
         multithreading       BULLET2_MULTITHREADING
         double-precision     USE_DOUBLE_PRECISION
-		extras               BUILD_EXTRAS
+        extras               BUILD_EXTRAS
     INVERTED_FEATURES
         rtti                 USE_MSVC_DISABLE_RTTI
 )

--- a/ports/bullet3/vcpkg.json
+++ b/ports/bullet3/vcpkg.json
@@ -19,14 +19,14 @@
     "double-precision": {
       "description": "Use float64 doubles for bullet3"
     },
+    "extras": {
+      "description": "Build the extras"
+    },
     "multithreading": {
       "description": "Multithreading functionality for bullet3"
     },
     "rtti": {
       "description": "Enable RTTI on windows"
-    },
-	"extras": {
-      "description": "Build the extras"
     }
   }
 }

--- a/ports/bullet3/vcpkg.json
+++ b/ports/bullet3/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "bullet3",
   "version": "3.25",
+  "port-version": 1,
   "description": "Bullet Physics is a professional collision detection, rigid body, and soft body dynamics library",
   "homepage": "https://github.com/bulletphysics/bullet3",
   "license": "Zlib",
@@ -23,6 +24,9 @@
     },
     "rtti": {
       "description": "Enable RTTI on windows"
+    },
+	"extras": {
+      "description": "Build the extras"
     }
   }
 }

--- a/versions/b-/bullet3.json
+++ b/versions/b-/bullet3.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "749b894e0161c23e525bf7aafbb43718d8fa26fa",
+      "git-tree": "97fa5e4897653fbef0b1084d71cf832c39726d60",
       "version": "3.25",
       "port-version": 1
     },

--- a/versions/b-/bullet3.json
+++ b/versions/b-/bullet3.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "749b894e0161c23e525bf7aafbb43718d8fa26fa",
+      "version": "3.25",
+      "port-version": 1
+    },
+    {
       "git-tree": "2ddb8da6a7bd04aff093231935f5ead2a8ee7c79",
       "version": "3.25",
       "port-version": 0

--- a/versions/b-/bullet3.json
+++ b/versions/b-/bullet3.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "97fa5e4897653fbef0b1084d71cf832c39726d60",
+      "git-tree": "9ecadb6b6e0651347e89555207f28c4ee5b1b51c",
       "version": "3.25",
       "port-version": 1
     },

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -1318,7 +1318,7 @@
     },
     "bullet3": {
       "baseline": "3.25",
-      "port-version": 0
+      "port-version": 1
     },
     "bustache": {
       "baseline": "1.1.0",

